lib/sass/script/functions.rb

```diff
@@ -2147,16 +2198,19 @@ def map_values(map)
     # @example
     #   map-has-key(("foo": 1, "bar": 2), "foo") => true
     #   map-has-key(("foo": 1, "bar": 2), "baz") => false
-    # @overload map_has_key($map, $key)
+    #   map-has-key((foo: (bar: 1)), foo, bar)   => true
+    # @overload map_has_key($map, $keys)
     #   @param $map [Sass::Script::Value::Map]
-    #   @param $key [Sass::Script::Value::Base]
+    #   @param $keys [[Sass::Script::Value::Base]]
     # @return [Sass::Script::Value::Bool]
     # @raise [ArgumentError] if `$map` is not a map
-    def map_has_key(map, key)
+    def map_has_key(map, *keys)
       assert_type map, :Map, :map
-      bool(map.to_h.has_key?(key))
+      bool(keys.all? do |key|
+        map = map.to_h[key] || false
+      end)
     end
-    declare :map_has_key, [:map, :key]
+    declare :map_has_key, [:map, :keys], :var_args => true
```

Comment:
First of all, I'd say I'm not 100% sure about named parameters. It's something I haven't really seen in the wild in Sass code, so I always forget that's a thing, so the idea of var_args declaring an "unnamed" variable seems uncomfortable to me.